src/System.Globalization.Calendars/tests/CalendarHelpers.cs

```diff
@@ -2,88 +2,213 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Collections.Generic;
+using System.Diagnostics;
 using Xunit;
 
 namespace System.Globalization.Tests
 {
     public static class CalendarHelpers
     {
-        private static int MaxEra(Calendar calendar) => calendar.GetEra(calendar.MaxSupportedDateTime);
+        public static Calendar[] s_calendars = new Calendar[]
+        {
+            new ThaiBuddhistCalendar(),
+            new JapaneseCalendar()
+        };
+
         private static int MinEra(Calendar calendar) => calendar.GetEra(calendar.MinSupportedDateTime);
+        private static int MaxEra(Calendar calendar) => calendar.GetEra(calendar.MaxSupportedDateTime);
 
-        private static int YearOffset(Calendar calendar) => calendar.GetYear(calendar.MinSupportedDateTime) - 1;
+        private static int YearOffset(Calendar calendar)
+        {
+            if (calendar is ThaiBuddhistCalendar)
+            {
+                return 543;
```

Comment:
I've tried to avoid hardcoding lots of values, but the only place I had to was YearOffset.
The previous implementation was incorrect. For example, KoreanCalendar has a MinSupportedDateTime value equal to DateTime.MinValue, but an offset of 543 years